spec/unit/util/windows/adsi_spec.rb

```diff
@@ -103,6 +103,25 @@
       expect(user.native_object).to eq(adsi_user)
     end
 
+    context "when domain-joined" do
```

Comment:
Missing same tests for `ADSI::Group`

Reply:
added them, although I would have preferred a less-duplicative way of doing it...

Reply:
Perhaps using Shared examples?

``` ruby
  RSpec.shared_examples 'a local only resource query' do |klass, resource_name, account_type|
    it "should be able to check for a local resource" do
      local_domain = 'testcomputername'
      principal = double('Principal', :account => resource_name, :domain => local_domain, :account_type => account_type)
      allow(Puppet::Util::Windows::SID).to receive(:name_to_principal).with(resource_name).and_return(principal)
      expect(klass.exists?(resource_name)).to eq(true)
    end

    it "should return false if no local resource exists" do
      principal = double('Principal', :account => resource_name, :domain => 'AD_DOMAIN', :account_type => account_type)
      allow(Puppet::Util::Windows::SID).to receive(:name_to_principal).with(resource_name).and_return(principal)
      expect(klass.exists?(resource_name)).to eq(false)
    end
  end
```

and then you can assert with:

``` ruby
    context "when domain-joined" do
      before(:each) do
        allow(Puppet::Util::Windows::ADSI).to receive(:domain_role).and_return(:MEMBER_SERVER)
      end

      it_should_behave_like 'a local only resource query', Puppet::Util::Windows::ADSI::Group, groupname, :SidTypeGroup
   end
```